crates/cargo-util/src/paths.rs

```diff
@@ -417,7 +417,10 @@ fn _create_dir_all(p: &Path) -> Result<()> {
 ///
 /// This does *not* follow symlinks.
 pub fn remove_dir_all<P: AsRef<Path>>(p: P) -> Result<()> {
-    _remove_dir_all(p.as_ref())
+    _remove_dir_all(p.as_ref()).or_else(|_| {
+        fs::remove_dir_all(p.as_ref())
+            .with_context(|| format!("failed to remove directory `{}`", p.as_ref().display()))
+    })
```

Comment:
The intention for the fallback isn't very clear here.  Could you document why this is needed in the closure?

We likely should somewhere document why this version of `remove_dir_all` even exists because that would be my first reaction to reading a comment of the lines "our version didn't work, so use the stdlib's version which is more thorough".

Reply:
Done. This version of `remove_dir_all` was added in https://github.com/rust-lang/cargo/pull/5110, and appears to have been done for improved error reporting.